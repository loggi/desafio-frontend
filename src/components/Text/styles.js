import styled from '@emotion/styled'
import { css } from '@emotion/react'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing(3)};
  `}
`
