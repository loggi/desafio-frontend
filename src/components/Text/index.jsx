import PropTypes from 'prop-types'

import * as S from './styles'

const Text = ({ tag = 'p', children, ...props }) => (
  <S.Wrapper as={tag} {...props}>
    {children}
  </S.Wrapper>
)

Text.propTypes = {
  tag: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Text
