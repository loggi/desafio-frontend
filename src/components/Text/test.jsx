import { screen } from '@testing-library/react'

import { render } from '@/utils/tests/helpers'

import Text from '.'

describe('<Text />', () => {
  it('should render the Text', () => {
    const { container } = render(<Text>Text</Text>)

    expect(screen.getByText(/Text/)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
