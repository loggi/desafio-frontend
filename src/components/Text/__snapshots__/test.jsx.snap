// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render the Text 1`] = `
.emotion-0 {
  margin-bottom: 24px;
}

<p
  class="emotion-0"
>
  Text
</p>
`;
